llama_stack/providers/remote/vector_io/qdrant/qdrant.py

```diff
@@ -134,7 +134,50 @@ async def query_keyword(
         k: int,
         score_threshold: float,
     ) -> QueryChunksResponse:
-        raise NotImplementedError("Keyword search is not supported in Qdrant")
+        """Perform keyword search using Qdrant's payload filtering capabilities."""
+        try:
+            results = (
+                await self.client.query_points(
+                    collection_name=self.collection_name,
+                    query_filter=models.Filter(
+                        must=[
+                            models.FieldCondition(
+                                key="chunk_content.content", match=models.MatchText(text=query_string)
+                            )
+                        ]
+                    ),
+                    limit=k,
+                    with_payload=True,
+                    with_vectors=False,
+                )
+            ).points
+
+            chunks, scores = [], []
+            for point in results:
+                assert isinstance(point, models.ScoredPoint)
+                assert point.payload is not None
+
+                try:
+                    chunk = Chunk(**point.payload["chunk_content"])
+                except Exception:
+                    log.exception("Failed to parse chunk")
+                    continue
+
+                content = chunk.content.lower()
+                query_lower = query_string.lower()
+                if query_lower in content:
+                    # Binary scoring: 1.0 for any match, 0.0 for no match
+                    score = 1.0
+
+                    if score >= score_threshold:
+                        chunks.append(chunk)
+                        scores.append(score)
+
+            return QueryChunksResponse(chunks=chunks, scores=scores)
+
+        except Exception as e:
```

Comment:
this pattern of adding very large `try-catch` blocks is not appropriate. you are essentially grouping all kinds of bad things -- including typos, basic logic mistakes, etc. -- all into a single error handler saying "error performing keyword search". That makes no sense. if there is an exception handler you are adding to the code it means "I don't know what this piece of code will do because it is somewhat outside of my control". Please rewrite appropriately.